### PR TITLE
CATROID-816 Basic Catblocks Integration

### PR DESCRIPTION
**Pull request by bprattes**
Add basic Catblocks 2D functionality to Catroid.
https://jira.catrob.at/browse/CATROID-816

### Your checklist for this pull request
Please review the [contributing guidelines](https://github.com/Catrobat/Catroid/blob/develop/README.md) and [wiki pages](https://github.com/Catrobat/Catroid/wiki/) of this repository.

- [x] Include the name of the Jira ticket in the PR’s title
- [x] Include a summary of the changes plus the relevant context
- [x] Choose the proper base branch (*develop*)
- [x] Confirm that the changes follow the project’s coding guidelines
- [x] Verify that the changes generate no compiler or linter warnings
- [x] Perform a self-review of the changes
- [x] Verify to commit no other files than the intentionally changed ones
- [x] Include reasonable and readable tests verifying the added or changed behavior
- [x] Confirm that new and existing unit tests pass locally
- [x] Check that the commits’ message style matches the [project’s guideline](https://github.com/Catrobat/Catroid/wiki/Commit-Message-Guidelines)
- [x] Stick to the project’s gitflow workflow
- [x] Verify that your changes do not have any conflicts with the base branch
- [x] After the PR, verify that all CI checks have passed
- [x] Post a message in the *catroid-stage* or *catroid-ide* [Slack channel](https://catrobat.slack.com) and ask for a code reviewer

### CODE REVIEW

**Review by MichaelFlucher — APPROVED** [review]
LGTM

**Review by wslany — CHANGES_REQUESTED** [review]
Awesome!

Please add a build config flag that enables the 2D view for now only in the develop version, since there are quite a number of issues. This shall be done as for instance shown here for the future multiplayer extension: https://github.com/Catrobat/Catroid/blob/13c5299201e84a08eb6fb6f503910b2c8f48e4e2/catroid/build.gradle#L166

Reason: We could not yet release the current version, since it allows to change the scripts in the 2D view (by moving parts around), which in principle is great, but these changes are currently still not kept in this version as soon as you switch back to the 1D view, so I guess the changes are just made in the Catblocks part, right? With the build config flag, we can merge Catblocks into develop but it is ignored for a release version (additional conditional code needs to be added of course for the flag). 

Also, by scrolling up you can currently switch the 2D script view to see the scripts of another object, but then tapping on the looks tab switches to the looks of the original object, thus not corresponding to the scripts. This is OK for now, but we need to fix it before we can release it, so OK to merge, but that's why we need the build config flag. 

image

Also, please fix the static analysis warnings: https://jenkins.catrob.at/job/Catroid/job/PR-3871/2/analysis/

**Review by wslany — APPROVED** [review]

**Review by wslany — CHANGES_REQUESTED** [review]
Please increase the Catrobat language version in https://github.com/Catrobat/Catroid/blob/develop/catroid/src/main/java/org/catrobat/catroid/common/Constants.java#L42 (please do not increase to 1.0 --- this is reserved to when the user-defined brick functionality is merged, which hopefully will be very soon).

File: catroid/build.gradle [hunk 1]
```diff
@@ -241,11 +243,14 @@ android {
...
             buildConfigField "boolean", "USE_ANDROID_LOCALES_FOR_SCREENSHOTS", useAndroidLocales()
             resValue "string", "DEBUG_MODE", "true"
+            buildConfigField "boolean", "FEATURE_CATBLOCKS_ENABLED", "true"
+            buildConfigField "boolean", "FEATURE_CATBLOCKS_DEBUGABLE", "true"
             testCoverageEnabled = rootProject.hasProperty('enableCoverage')
             signingConfig signingConfigs.debug
         }
 
         release {
+            buildConfigField "boolean", "FEATURE_CATBLOCKS_DEBUGABLE", "false"
```
Comment (juliajulie95) [inline catroid/build.gradle:253]: Delete this if already set by default config

File: catroid/src/main/java/org/catrobat/catroid/ui/recyclerview/fragment/ScriptFragment.java [hunk 2]
```diff
@@ -263,6 +277,10 @@ public void onPause() {
 	public void onPrepareOptionsMenu(Menu menu) {
 		menu.findItem(R.id.show_details).setVisible(false);
 		menu.findItem(R.id.rename).setVisible(false);
+		menu.findItem(R.id.catblocks_reorder_scripts).setVisible(false);
```
Comment (juliajulie95) [inline catroid/src/main/java/org/catrobat/catroid/ui/recyclerview/fragment/ScriptFragment.java:280]: Hide action mode in looks and sounds

File: catroid/src/main/java/org/catrobat/catroid/ui/recyclerview/fragment/CatblocksScriptFragment.kt [hunk 3]
```diff
@@ -0,0 +1,252 @@
...
+        menu.removeItem(R.id.delete)
+        menu.removeItem(R.id.rename)
+        menu.removeItem(R.id.show_details)
+    }
+
+    override fun onOptionsItemSelected(item: MenuItem): Boolean {
+        if (item.itemId == R.id.catblocks) {
+            SettingsFragment.setUseCatBlocks(context, false)
+
+            val fragmentTransaction = fragmentManager?.beginTransaction()
```
Comment (juliajulie95) [inline catroid/src/main/java/org/catrobat/catroid/ui/recyclerview/fragment/CatblocksScriptFragment.kt:81]: deprecated, use getParentFragmentManager() if possible

File: catroid/src/main/java/org/catrobat/catroid/ui/recyclerview/fragment/CatblocksScriptFragment.kt [hunk 4]
```diff
@@ -0,0 +1,252 @@
...
+
+        val view = View.inflate(activity, R.layout.fragment_catblocks, null)
+        val webView = view.findViewById<WebView>(R.id.catblocksWebView)
+        initWebView(webView)
+        this.webview = webView
+        return view
+    }
+
+    @SuppressLint("SetJavaScriptEnabled")
+    private fun initWebView(catblocksWebview: WebView) {
```
Comment (juliajulie95) [inline catroid/src/main/java/org/catrobat/catroid/ui/recyclerview/fragment/CatblocksScriptFragment.kt:122]: Naming

File: catroid/src/main/java/org/catrobat/catroid/ui/recyclerview/fragment/CatblocksScriptFragment.kt [hunk 5]
```diff
@@ -0,0 +1,252 @@
...
+            JSInterface(
+                currentProject,
+                currentScene,
+                currentSprite,
+                currentScriptIndex
+            ), "Android"
+        )
+
+        catblocksWebview.webViewClient = object : WebViewClient() {
+            @RequiresApi(Build.VERSION_CODES.LOLLIPOP)
```
Comment (juliajulie95) [inline catroid/src/main/java/org/catrobat/catroid/ui/recyclerview/fragment/CatblocksScriptFragment.kt:144]: Do we need this? Else disable catblocks for lower android versions

**Review by juliajulie95 — CHANGES_REQUESTED** [review]

**Review by juliajulie95 — CHANGES_REQUESTED** [review]
Please delete unused imports.
Everything else looks fine

**Review by juliajulie95 — APPROVED** [review]

**Review by wslany — APPROVED** [review]

**Comment by wslany** [comment]
Thanks!! Awesome!

**Pull request merged**